mailpoet/assets/js/src/wizard/steps/pitch_mss_step/second_part.tsx

```diff
@@ -0,0 +1,76 @@
+import { useHistory, useParams } from 'react-router-dom';
+import ReactStringReplace from 'react-string-replace';
+import { MailPoet } from 'mailpoet';
+import { Heading } from 'common';
+import { KeyActivationButton } from 'common/premium_key/key_activation_button';
+import { KeyInput } from 'common/premium_key/key_input';
+import { useEffect } from 'react';
+import { useSelector } from 'settings/store/hooks';
+import { OwnEmailServiceNote } from './own_email_service_note';
+
+function MSSStepSecondPart(): JSX.Element {
+  const history = useHistory();
+  const { step } = useParams<{ step: string }>();
+  const state = useSelector('getKeyActivationState')();
+
+  useEffect(() => {
```

Comment:
I'm unsure if using `useEffect()` here is the best way to change to the next part of this step when the user adds a valid key. This has the downside that the code runs when the components mount for the first time which is not necessary in this case.

Reply:
This is interesting. We have a state in store and steps state in route 🤔 I don't know what is the best practice synchronizing this. I think we can go with this solution.

Reply:
Ok, thanks for checking.